docs/tp/03.md

```diff
@@ -0,0 +1,208 @@
+
+```rust
+use std::fs;
+```
+### Le struct File
+Le struct `File` renvoie un objet donnant accès à un fichier ouvert sur le système de fichiers.
+
+Une instance d'un fichier peut être lue et/ou écrite en fonction des **options* avec lesquelles elle a été ouverte. Les fichiers implémentent également Seek pour modifier le curseur logique que le fichier contient en interne.
+
+Les fichiers sont **automatiquement fermés** lorsqu'ils sortent de **scope**. Les erreurs détectées à la fermeture sont ignorées par l'implémentation de `Drop`. Utilisez la méthode `sync_all` si ces erreurs doivent être gérées manuellement.
+
+### Créer un fichier
+La fonction `File::create()` ouvre un fichier en mode **read-only**.
+
+Cette fonction créera un fichier s'il n'existe pas et le tronquera s'il existe.
+
+Selon la plate-forme, cette fonction peut échouer si **le chemin d'accès** complet au répertoire n'existe pas.
+
+```rust
+use std::fs::File;
+ 
+fn main() -> std::io::Result<()> {
+    let mut f = File::create("foo.txt")?;
+    Ok(())
+}
+```
+
+### Ouvrir un fichier existant
+La fonction `File::open()` tente d'ouvrir un fichier en mode read-only.
+
+```rust
+use std::fs::File;
+ 
+fn main() -> std::io::Result<()> {
+    let mut f = File::open("foo.txt")?;
+    Ok(())
+}
+```
+### Lire le contenu d'un fichier
+Pour lire à partir d'un fichier, nous utilisons la fonction `read_to_string(file_path)`:
+
+```rust
+use std::fs;
+use std::env;
+ 
+fn main() {
+ 
+    println!("In file {}", file_path);
+ 
+    let contents = fs::read_to_string(file_path)
+        .expect("Should have been able to read the file");
+ 
+    println!("With text:\n{contents}");
+}
+```
+
+### Le struct Metadata
+Cette struct renvoie informations de métadonnées sur un fichier.
+
+Ce module contient des méthodes pour récupérer des informations sur un fichier telles que le type de fichier, les autorisations, etc.
+
+:::note
+Pour consulter toutes les méthodes disponibles, veuillez lire la documentation suivante: [Metadata crate](https://doc.rust-lang.org/stable/std/fs/struct.Metadata.html)!
+:::
+
+### Le descripteur de fichier
+Un **descripteur de fichier** est un numéro qui identifie de manière unique un fichier ouvert dans le système d'exploitation d'un ordinateur. Il décrit une ressource de données et comment cette ressource peut être accessible.
+
+Lorsqu'un programme demande d'ouvrir un fichier **le kernel** fait les actions suivantes:
+1. Accorde l'accès
+2. Crée une entrée dans la table de fichiers globale
+3. Fournit au logiciel l'emplacement de cette entrée
+
+Le descripteur est identifié par un **entier non négatif unique**, tel que 0, 12 ou 567. Au moins un descripteur de fichier existe pour chaque fichier ouvert sur le système.
+
+Pour obtenir le file descriptor d'un fichier on utilise la fonction [as_raw_fd()](https://doc.rust-lang.org/stable/std/os/fd/trait.AsRawFd.html#tymethod.as_raw_fd).
+
+## Le type Result
```

Comment:
I would ,ove this before the functions for files, as those functions use it.

Reply:
Done

Reply:
We still have `Result` for the main function, maybe move it ahead of that?